Energy Saver/Pages/Account/Profile.cshtml.cs

```diff
@@ -42,31 +50,57 @@ public void OnGet()
             UserProfileImage = User.FindFirst(c => c.Type == "picture")?.Value;
         }
 
-        public IActionResult OnPost()
+        public async Task<IActionResult> OnPostDetails()
         {
             var userId = User.FindFirst(ClaimTypes.NameIdentifier)?.Value;
+            var hasUpdatedName = await _profileService.UpdateUserNameAsync(userId, UserName);
+            var hasUpdatedEmail = await _profileService.UpdateUserEmailAsync(userId, UserEmailAddress);
 
-            var client = new RestClient("https://priolette.eu.auth0.com/api/v2/");
-            var request = new RestRequest($"/users/{userId}", RestSharp.Method.Patch);
+            if (hasUpdatedEmail && hasUpdatedName)
+                OnProfileEditSuccess();
+            else
+                OnProfileEditError("Could not update user profile");
 
-            request.AddHeader("content-type", "application/json");
-            request.AddHeader("authorization", $"Bearer {_config["Auth0ApiToken"]}");
-            request.AddHeader("cache-control", "no-cache");
-            request.AddParameter("application/json", $"{{\"nickname\": \"{UserName}\" }}", ParameterType.RequestBody);
+            return RedirectToPage("./Profile");
 
-            RestResponse response = client.Execute(request);
-            _logger.LogInformation(response.Content);
+            //OLD IMPLEMENTATION (DO NOT REMOVE YET):
 
-            OnProfileEditSuccess();
+            //var client = new RestClient("https://priolette.eu.auth0.com/api/v2/");
+            //var request = new RestRequest($"/users/{userId}", RestSharp.Method.Patch);
 
-            return RedirectToPage("./Profile");
+            //request.AddHeader("content-type", "application/json");
```

Comment:
Will you still work with these commented lines? If not, then they should probably be deleted

Reply:
i might still need them in the future for a few things, so i'll keep them right now